### PR TITLE
Add the capability to resolve timezone information regardless of hte platform and database being used.

### PR DESCRIPTION
**Pull request by SeanFarrow**
…en windows and the iana database used on most other systems."

Complete the implementeation of the plugin allowing resolution of timezones regardless of the platform and database being used.

### CODE REVIEW

File: Quartz.sln [hunk 1]
```diff
@@ -25,11 +25,13 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Quartz.Examples", "src\Quar
...
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Quartz.Server", "src\Quartz.Server\Quartz.Server.csproj", "{09082B9A-906C-4A17-A2E5-6C947DAC7C85}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Quartz.Server", "src\Quartz.Server\Quartz.Server.csproj", "{09082B9A-906C-4A17-A2E5-6C947DAC7C85}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Quartz.Jobs", "src\Quartz.Jobs\Quartz.Jobs.csproj", "{5D31C178-7349-4064-970A-223B45E913C1}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Quartz.Jobs", "src\Quartz.Jobs\Quartz.Jobs.csproj", "{5D31C178-7349-4064-970A-223B45E913C1}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Quartz.Plugins", "src\Quartz.Plugins\Quartz.Plugins.csproj", "{E9A60600-156A-4F3C-B080-B630D6546164}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Quartz.Plugins", "src\Quartz.Plugins\Quartz.Plugins.csproj", "{E9A60600-156A-4F3C-B080-B630D6546164}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Quartz.Plugins.TimezoneResolution", "src\Quartz.Plugins.TimezoneResolution\Quartz.Plugins.TimezoneResolution.csproj", "{88F6D094-A4CD-4B89-84FB-FBD0821A9D99}"
```
Comment (lahma) [inline Quartz.sln]: I think we should refer this to TimeZoneResolution as .NET BCL has TimeZoneInfo, so casing would be similar

File: src/Quartz.Plugins.TimezoneResolution/Plugin/TimezoneResolution/TimezoneResolutionPlugin.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        public Task Initialize(
+            string pluginName, 
+            IScheduler scheduler, 
+            CancellationToken cancellationToken = default)
+        {
+            // do nothing...
+            return TaskUtil.CompletedTask;
+        }
+
+        /// <summary>
```
Comment (lahma) [inline src/Quartz.Plugins.TimezoneResolution/Plugin/TimezoneResolution/TimezoneResolutionPlugin.cs]: indentation is a bit off for doc comment and method declaration

File: src/Quartz.Plugins.TimezoneResolution/Plugin/TimezoneResolution/TimezoneResolutionPlugin.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            /// Called when the associated <see cref="IScheduler" /> is started, in order
+            /// to let the plug-in know it can now make calls into the scheduler if it
+            /// needs to.
+            /// </summary>
+public Task Start(CancellationToken cancellationToken = default)
+        {
+            TimeZoneUtil.CustomResolver = timeZoneID => TZConvert.GetTimeZoneInfo(timeZoneID);
```
Comment (lahma) [inline src/Quartz.Plugins.TimezoneResolution/Plugin/TimezoneResolution/TimezoneResolutionPlugin.cs]: should this be actually in initialize to get available as soon as possible?

**Comment by SeanFarrow** [comment]
Hi Marko,

Can you fix the indentation, I’m totally blind, so wouldn’t have spotted this!
What extra details do we want in the project, it seem descriptive enough!

**Comment by SeanFarrow** [comment]
Possibly, but can any resolution happen before start?

**Comment by lahma** [comment]
Thanks for the contribution! I'll add a small tweak on top of this, small rename and other tweaks mentioned.

**Pull request merged**